crates/task-impls/src/transactions.rs

```diff
@@ -231,25 +262,38 @@ impl<
                 continue;
             };
 
-            let mut available_blocks = match self
-                .builder_client
-                .get_available_blocks(
+            let mut available_blocks = match async_compatibility_layer::art::async_timeout(
+                self.api.propose_max_round_time(),
+                self.builder_client.get_available_blocks(
                     parent_commitment,
                     self.public_key.clone(),
                     &request_signature,
-                )
-                .await
+                ),
+            )
+            .await
             {
-                Ok(blocks) => {
+                // We got available blocks
+                Ok(Ok(blocks)) => {
                     tracing::debug!("Got available blocks: {:?}", blocks);
                     blocks
                 }
-                Err(err) => {
+
+                // We failed to get available blocks
+                Ok(Err(err)) => {
                     error!(%err, "Couldn't get available blocks");
                     // pause a bit
                     async_sleep(Duration::from_millis(100)).await;
                     continue;
                 }
+
+                // We timed out while getting available blocks
+                Err(err) => {
+                    error!(%err, "Timeout while getting available blocks");
+                    // pause a bit
+                    async_sleep(Duration::from_millis(100)).await;
```

Comment:
Not sure I understand this bit. If timeout is `propose_max_round_time()`, aren't we are out of time and need to propose whatever we have ASAP, not try again?

Reply:
Agree, I think just `return None;` here

Reply:
That makes sense, this was to just a stopgap to prevent us fromn being dependent on the builder, which can hang sometimes

Reply:
> Agree, I think just `return None;` here

We may already have something in `latest_block`, should be `break;`